packages/EasyCodingStandard/src/Finder/DefaultFilesProvider.php

```diff
@@ -0,0 +1,54 @@
+<?php declare(strict_types=1);
+
+namespace Symplify\EasyCodingStandard\Finder;
+
+use SplFileInfo;
+use Symfony\Component\Finder\Finder;
+use Symplify\EasyCodingStandard\Contract\Finder\ExtraFilesProviderInterface;
+
+final class DefaultFilesProvider implements ExtraFilesProviderInterface
```

Comment:
Could you return this method return to `SourceFinder`? This creates confusion, doesn't fit current architecture and I don't see a way to solve it nicely without many changes.

You can keep a check that you added before:

```php
if (! count($this->extraFilesProvider)) {
    return $this->findPhpFilesInSource($source);
}

foreach ($this->extraFilesProvider as $extraFilesProvider) {
    $files = array_merge($files, $extraFilesProvider->provideForSource($source));
}

return $files;
```